### PR TITLE
ui: create filter component

### LINKED ISSUE #4073: [Subtask]: Create FilterComponent

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary
Create a filter framework that can be used to build filters for different types of lists

## Changelog

```
- create generic filter component
- create simple boolean filter
- asset filter
- activity filter
```

## Relevant Issues
Closes #4073 

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/locales/en.json [hunk 1]
```diff
@@ -1271,6 +1272,15 @@
...
 
     },
+    "filters":{
+        "title": "Filters",
+        "assets": {
+            "showHidden": "Show hidden"
+        },
+        "activity": {
+            "showHidden": "Show hidden"
+        }
```
Comment (Tuditi) [inline packages/shared/locales/en.json:1282]: Maybe this could be one locale only?

File: packages/shared/components/filters/AssetFilter.svelte [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {#if filterActive}
+        <div class="absolute right-60 top-30">
+            <FilterModal bind:modal bind:filter {onSetFilters}>
+                {#each Object.keys(filter) as filterItem}
+                    <FilterItem filterItem={filter[filterItem]} />
+                {/each}
+            </FilterModal>
+        </div>
+    {/if}
+</div>
```
Comment (Tuditi) [inline packages/shared/components/filters/AssetFilter.svelte:32]: Looks like this is a duplication of `ActivityFilter`, maybe it's possible to use generics for the types and create a `Filter` oganism that is used by both?

Comment (MarkNerdi) [reply packages/shared/components/filters/AssetFilter.svelte]: Good point, implemented it in the newest commits

File: packages/shared/locales/en.json (hunk 1 above)
Comment (MarkNerdi) [reply packages/shared/locales/en.json:1282]: there won't be any overlaps on the other filters that will come in the next branches. So i would add each filter for each filtertype on the corresponding locale object and therefore let this duplicate

File: packages/shared/components/filters/Filter.svelte [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    export let filterStore: Writable<Filter>
+
+    let filter: Filter = deepCopy($filterStore)
+
+    let filterActive = false
+    let modal: Modal
+
+    $: filterActive && modal?.toggle()
+
+    function onSetFilters() {
```
Comment (Tuditi) [inline packages/shared/components/filters/Filter.svelte:16]: Missing return type

File: packages/shared/components/filters/Filter.svelte [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        $filterStore = deepCopy(filter)
+        filterActive = false
+    }
+
+    function onClose() {
```
Comment (Tuditi) [inline packages/shared/components/filters/Filter.svelte:21]: Missing return type

File: packages/shared/components/inputs/Checkbox.svelte [hunk 5]
```diff
@@ -21,10 +23,12 @@
 >
     <Icon
         icon={checked ? `checkbox${round ? '-round' : ''}` : `checkbox-unchecked${round ? '-round' : ''}`}
+        width={small ? '16px' : undefined}
+        height={small ? '16px' : undefined}
         classes={`${label ? 'mr-3' : ''} ${checked ? 'active' : ''}`}
     />
     {#if label}
-        <Text type="p" secondary={!checked || disabled}>{label}</Text>
+        <Text type="p" secondary={!keepSameColor && (!checked || disabled)}>{label}</Text>
```
Comment (Tuditi) [inline packages/shared/components/inputs/Checkbox.svelte:31]: Maybe the `secondary` condition could be named?

File: packages/shared/components/modals/FilterModal.svelte [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    import { Modal, Text } from 'shared/components'
+    import { Filter } from '@core/wallet/interfaces/filter.interface'
+    import { localize } from '@core/i18n'
+
+    export let modal: Modal
+    export let filter: Filter
+    export let onSetFilters: () => void
+    export let onClose: () => void
+
+    function clear() {
```
Comment (Tuditi) [inline packages/shared/components/modals/FilterModal.svelte:12]: Missing return type

File: packages/shared/components/modals/FilterModal.svelte [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        for (const key in filter) {
+            filter[key].active = false
+            filter[key].value = undefined
+        }
+        onSetFilters()
+        modal.toggle()
+    }
+
+    function confirm() {
```
Comment (Tuditi) [inline packages/shared/components/modals/FilterModal.svelte:21]: missing return type

File: packages/shared/lib/core/wallet/utils/isFilteredActivity.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { get } from 'svelte/store'
+import { Activity } from '../classes'
+import { activityFilter } from '../stores'
+
+export function isFilteredActivity(activity: Activity): boolean {
+    const filter = get(activityFilter)
+
+    if (!filter.showHidden.active && filter.showHidden.active && activity.isAssetHidden) {
```
Comment (Tuditi) [inline packages/shared/lib/core/wallet/utils/isFilteredActivity.ts:8]: Arent' the first two conditions contradictory?

File: packages/shared/locales/en.json (hunk 1 above)
Comment (Tuditi) [reply packages/shared/locales/en.json:1282]: I would argue that it's generic for a filter to show hidden items, therefor at the moment we don't need the two branches (assets & activity). We can add branches when we clearly need to distinguish some filter locales based on filter type.

**Review by Tuditi — COMMENTED** [review]
Couple of minor issues to resolve, but looks fine in general.

File: packages/shared/components/filters/FilterItem.svelte [hunk 9]
```diff
@@ -1,9 +1,9 @@
 <script lang="typescript">
-    import { ValueFilter } from '@core/wallet/interfaces/filter.interface'
+    import { FilterUnit } from '@core/wallet/interfaces/filter.interface'
     import { Checkbox } from 'shared/components'
     import { localize } from '@core/i18n'
 
-    export let filterItem: ValueFilter
+    export let filterItem: FilterUnit
```
Comment (Tuditi) [inline packages/shared/components/filters/FilterItem.svelte:6]: Maybe rename this to `filterUnit`

**Review by Tuditi — APPROVED** [review]
One suggestion, but LGTM otherwise!

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Icons need better alignment, some have different heights and positioning.
Screenshot 2022-08-01 at 17 32 52

**Comment by nicole-obrien** [comment]
Style of the component:
Screenshot 2022-08-02 at 16 10 41

Doesn't match style of design:
Screenshot 2022-08-02 at 16 11 27

**Review by nicole-obrien — APPROVED** [review]
Approving and we need to create a task for fixing the button styles. I understand the button component is not very great to work with and we need to refactor it.

**Pull request merged**